internal/.otel_collector_mixin/receiver/otlpreceiver/mixin.go

```diff
@@ -48,24 +51,72 @@ var settings = component.ReceiverCreateSettings{
 	},
 }
 
-// RegisterTraceReceiver registers the trace receiver with a gRPC server.
-func RegisterTraceReceiver(ctx context.Context, consumer consumer.Traces, serverGRPC *grpc.Server) error {
+type HTTPReceivers struct {
+	TracesR  *trace.Receiver
+	MetricsR *metrics.Receiver
+	LogsR    *logs.Receiver
+}
+
+// GRPC Receivers
 
+// RegisterGRPCTraceReceiver registers the trace receiver with a gRPC server.
+func RegisterGRPCTraceReceiver(ctx context.Context, consumer consumer.Traces, serverGRPC *grpc.Server) error {
 	receiver := trace.New(config.NewComponentID("otlp"), consumer, settings)
 	otlpgrpc.RegisterTracesServer(serverGRPC, receiver)
 	return nil
 }
 
-// RegisterMetricsReceiver registers the metrics receiver with a gRPC server.
-func RegisterMetricsReceiver(ctx context.Context, consumer consumer.Metrics, serverGRPC *grpc.Server) error {
+// RegisterGRPCMetricsReceiver registers the metrics receiver with a gRPC server.
+func RegisterGRPCMetricsReceiver(ctx context.Context, consumer consumer.Metrics, serverGRPC *grpc.Server) error {
 	receiver := metrics.New(config.NewComponentID("otlp"), consumer, settings)
 	otlpgrpc.RegisterMetricsServer(serverGRPC, receiver)
 	return nil
 }
 
-// RegisterLogsReceiver registers the logs receiver with a gRPC server.
-func RegisterLogsReceiver(ctx context.Context, consumer consumer.Logs, serverGRPC *grpc.Server) error {
+// RegisterGRPCLogsReceiver registers the logs receiver with a gRPC server.
+func RegisterGRPCLogsReceiver(ctx context.Context, consumer consumer.Logs, serverGRPC *grpc.Server) error {
 	receiver := logs.New(config.NewComponentID("otlp"), consumer, settings)
 	otlpgrpc.RegisterLogsServer(serverGRPC, receiver)
 	return nil
 }
+
+// HTTP Receivers
+
+func NewHTTPTraceReceiver(ctx context.Context, consumer consumer.Traces) (*trace.Receiver, error) {
+	receiver := trace.New(config.NewComponentID("otlp"), consumer, settings)
+	if consumer == nil {
+		return nil, componenterror.ErrNilNextConsumer
+	}
+
+	return receiver, nil
+}
+
+func HandleHTTPTraces(resp http.ResponseWriter, req *http.Request, traceReceiver *trace.Receiver) {
+	handleTraces(resp, req, traceReceiver, pbEncoder)
+}
```

Comment:
Rather than exposing the TraceReceiver and passing it back into HandleHTTPFoo, how about we have functions that construct handlers given a consumer? e.g.

```go
func TracesHTTPHandler(consumer consumer.Traces) http.Handler {
    receiver := trace.New(config.NewComponentID("otlp"), consumer, settings)
    return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
        handleTraces(w, r, traceReceiver, pbEncoder)
    })
}
```

Reply:
Implemented in 2f6d307fc